From 0e58add7fe30cfbda7295e0f617f22a30a6c4a06 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Mon, 3 Nov 2014 07:52:17 -0500
Subject: [infiniband] iser: Print QP information once connection is established

Message-id: <cfe0e93f285eb7c53a8f1658c88c30bfe2b5e59b.1415000301.git.avadai@redhat.com>
Patchwork-id: 98688
O-Subject: [Patch RHEL7.1 19/37] IB/iser: Print QP information once connection is established
Bugzilla: 1107622
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Doug Ledford <dledford@redhat.com>

From: Or Gerlitz <ogerlitz@mellanox.com>

Add an iser info print with the local/remote QP information carried
out when the connection is established.  While here, fix a little
leftover from the T10 work and set a debug print to be carried in
debug and not info level.

Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 4f9208ad3f6625f707210c00c7e7f7a0f688d019)
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/ulp/iser/iser_verbs.c b/drivers/infiniband/ulp/iser/iser_verbs.c
index b3668a5..b6ff08e 100644
--- a/drivers/infiniband/ulp/iser/iser_verbs.c
+++ b/drivers/infiniband/ulp/iser/iser_verbs.c
@@ -340,8 +340,8 @@ iser_create_fastreg_desc(struct ib_device *ib_device, struct ib_pd *pd,
  }
  desc->reg_indicators &= ~ISER_FASTREG_PROTECTED;
 
- iser_info("Create fr_desc %p page_list %p\n",
-    desc, desc->data_frpl->page_list);
+ iser_dbg("Create fr_desc %p page_list %p\n",
+   desc, desc->data_frpl->page_list);
 
  return 0;
 sig_mr_failure:
@@ -728,6 +728,11 @@ failure:
 static void iser_connected_handler(struct rdma_cm_id *cma_id)
 {
  struct iser_conn *ib_conn;
+ struct ib_qp_attr attr;
+ struct ib_qp_init_attr init_attr;
+
+ (void)ib_query_qp(cma_id->qp, &attr, ~0, &init_attr);
+ iser_info("remote qpn:%x my qpn:%x\n", attr.dest_qp_num, cma_id->qp->qp_num);
 
  ib_conn = (struct iser_conn *)cma_id->context;
  ib_conn->state = ISER_CONN_UP;
-- 
1.7.1